### ISSUE TITLE
Restrict cancelling of schedules of messages already sent

### ISSUE THREAD

**Issue by lkwirica** [intrahealth/emNutt#63]
![image](https://user-images.githubusercontent.com/77771356/122706879-8336cf00-d261-11eb-889d-8a57d31373d4.png)
![image](https://user-images.githubusercontent.com/77771356/122706890-8b8f0a00-d261-11eb-9c68-7479c768d848.png)

Related to issue #7

**Comment by ashaban** [comment]
@lkwirica can you give more details about this issue?

**Comment by lkwirica** [comment]
For integrity purposes, one can only cancel the record of a scheduled message BEFORE the message is sent out. Once the message has been sent, the record should not be deleted....This way, we can tell, all  historical message schedules in the system

**Comment by ashaban** [comment]
@lkwirica This report only shows Active Schedules, Cancelling a schedule doesnt delete, it just deactivate. If you need a report of all scheduled messages we can then add a new report that list all scheduled messages (active and inactive), but not this one

**Comment by cloehr651** [comment]
@lkwirica if you would like a new report please create a new issue that details what you want that report to include

**Comment by lkwirica** [comment]
@ashaban 'Active Schedule' in this case means a schedule whose message has been sent out/ not sent?

The schedules displayed are of messages already sent out....Shouldn't it then display schedules Not yet sent out? As in yet to be activated?

**Comment by ashaban** [comment]
Active Schedules is any schedule whose messages can still be sent out, remember that you can schedule a message to be sent out repeatedly i.e evey Monday and Tuesday, inactive schedule is the one that has either been cancelled or a schedule that can no longer be executed i.e you create a schedule to be executed twice in July, this schedule will be inactive in August.
So the report is displaying active schedules to allow you to cancel any schedule you think it should not be executed or no longer needed to run.

**Comment by lkwirica** [comment]
![image](https://user-images.githubusercontent.com/77771356/123031133-798da280-d3ec-11eb-8f99-85803335b2a8.png)
@ashaban Since the message was only to be sent out once, and the date passed, this should not be displaying on the 'Scheduled Messages' page...

**Comment by ashaban** [comment]
@lkwirica  Yes, it should not, for now that feature is not supported, but you can still manually take them out by cancelling them. We will later add auto cancelling of schedules that has passed their time of execution, but this is not a priority for now since you can still manually cancel schedules.
